packages/model-client/src/index.ts

```diff
@@ -101,6 +183,87 @@ export class ModelClient extends StreamClient {
     return getModelStreamID(cid)
   }
 
+  /**
+   * Posts a data event to a model stream and returns its commit ID.
+   *
+   * @param params - Parameters for posting the data event.
+   * @returns A promise that resolves to the `CommitID` of the posted event.
+   *
+   * @remarks
+   * The data event updates the content of a stream and is associated with the
+   * current state of the stream.
+   */
+  async postDefinition(params: PostDefinitionParams): Promise<CommitID> {
+    const { controller, ...rest } = params
+    const event = await createDataEvent({
+      ...rest,
+      controller: this.getDID(controller),
+    })
+    const cid = await this.ceramic.postEventType(SignedEvent, event)
+    return CommitID.fromStream(params.currentID.baseID, cid)
+  }
+
+  /**
+   * Updates a model with a new definition and returns the updated model state.
+   * Model's can only be updated in backwards compatible ways.
+   *
+   * @param params - Parameters for updating the document.
+   * @returns A promise that resolves to the updated `ModelState`.
+   *
+   * @remarks
+   * This method posts the new content as a data event, updating the document.
+   * It can optionally take the current document state to avoid re-fetching it.
+   */
+  async updateDefinition(
+    params: UpdateModelDefinitionParams,
+  ): Promise<ModelState> {
+    let currentState: ModelState
+    let currentId: CommitID
+
+    if (!params.currentState) {
+      const streamState = await this.getStreamState(
+        StreamID.fromString(params.streamID),
+      )
+      currentState = this.streamStateToModelState(streamState)
+      currentId = this.getCurrentID(streamState.event_cid)
+    } else {
+      currentState = this.streamStateToModelState(params.currentState)
+      currentId = this.getCurrentID(params.currentState.event_cid)
+    }
+
+    const { content } = currentState
+    const { controller, newContent } = params
+
+    const newCommit = await this.postDefinition({
+      controller: this.getDID(controller),
+      currentContent: content ?? undefined,
+      newContent,
+      currentID: currentId,
+    })
+
+    return {
+      commitID: newCommit,
+      content: newContent,
+      metadata: {
+        model: currentState.metadata.model,
+        controller: currentState.metadata.controller,
+        ...(typeof currentState.metadata === 'object'
+          ? currentState.metadata
+          : {}),
+      },
+    }
+  }
+
+  /**
+   * Retrieves the `CommitID` for the provided stream ID.
+   *
+   * @param streamID - The stream ID string.
+   * @returns The `CommitID` for the stream.
+   */
+  getCurrentID(streamID: string): CommitID {
+    return new CommitID(2, streamID)
```

Comment:
What does this do?

Reply:
constructs a CommitID from just a stream id. Therefore the commit referenced will be the init commit